File: cplus/leetcode/202603/algo3138.cpp
#include <bits/stdc++.h>
using namespace std;

// 顺序做题
class Solution {
public:
    int minAnagramLength(string s) {
        // 自行解答
        int n = s.length();
        for (int l=1; l<n; l++) {
            if (n % l) continue;
            string first = s.substr(0, l); sort(first.begin(), first.end());

            bool found = true;
            for (int i=l; i<n; i+=l) {
                string other = s.substr(i, l); sort(other.begin(), other.end());
                if (first != other) {
                    found = false; break;
                }
            }

            if (found) return l;
        }
        return n;
    }
}; 

File: python/algo/202603/algo3138.py
class Solution:
    def minAnagramLength(self, s: str) -> int:
        # 自行解答， for ... else
        n = len(s)

        for l in range(1, n):
            if n % l: continue

            first = sorted(s[0:l])

            for i in range(l, n, l) :
                other = sorted(s[i:i+l])
                
                if (other!=first): break
            else: return l
        return n
